webpack.config.js

```diff
@@ -79,192 +72,154 @@ const base = {
+    }))
+    .addPlugin(new webpack.DefinePlugin({
+        'process.env.DEBUG': Boolean(process.env.DEBUG),
+        'process.env.GA_ID': `"${process.env.GA_ID || 'UA-000000-01'}"`,
+        'process.env.GTM_ENV_AUTH': `"${process.env.GTM_ENV_AUTH || ''}"`,
+        'process.env.GTM_ID': process.env.GTM_ID ? `"${process.env.GTM_ID}"` : null
+    }))
+    .addPlugin(new CopyWebpackPlugin({
+        patterns: [
+            {
+                from: 'node_modules/scratch-blocks/media',
+                to: 'static/blocks-media/default'
+            },
+            {
+                from: 'node_modules/scratch-blocks/media',
+                to: 'static/blocks-media/high-contrast'
+            },
+            {
+                // overwrite some of the default block media with high-contrast versions
+                // this entry must come after copying scratch-blocks/media into the high-contrast directory
+                from: 'src/lib/themes/high-contrast/blocks-media',
+                to: 'static/blocks-media/high-contrast',
+                force: true
+            },
+            {
+                context: 'node_modules/scratch-vm/dist/web',
+                from: 'extension-worker.{js,js.map}',
+                noErrorOnMissing: true
+            }
+        ]
+    }));
 
 if (!process.env.CI) {
-    base.plugins.push(new webpack.ProgressPlugin());
+    baseConfig.addPlugin(new webpack.ProgressPlugin());
 }
 
-module.exports = [
-    // to run editor examples
-    defaultsDeep({}, base, {
+// build the shipping library in `dist/`
+const distConfig = baseConfig.clone()
+    .merge({
         entry: {
-            'lib.min': ['react', 'react-dom'],
-            'gui': './src/playground/index.jsx',
-            'blocksonly': './src/playground/blocks-only.jsx',
-            'compatibilitytesting': './src/playground/compatibility-testing.jsx',
-            'player': './src/playground/player.jsx'
+            'scratch-gui': path.join(__dirname, 'src/index.js')
         },
         output: {
-            path: path.resolve(__dirname, 'build'),
-            filename: '[name].js'
-        },
-        module: {
-            rules: base.module.rules.concat([
-                {
-                    test: /\.(svg|png|wav|mp3|gif|jpg)$/,
-                    loader: 'url-loader',
-                    options: {
-                        limit: 2048,
-                        outputPath: 'static/assets/'
-                    }
-                }
-            ])
-        },
-        optimization: {
-            splitChunks: {
-                chunks: 'all',
-                name: 'lib.min'
+            path: path.resolve(__dirname, 'dist')
+        }
+    });
+
+// build the examples and debugging tools in `build/`
+const buildConfig = baseConfig.clone()
+    .merge({
+        devServer: {
+            client: {
+                progress: true
             },
-            runtimeChunk: {
-                name: 'lib.min'
-            }
```

Comment:
The previous version expressed more control over the way webpack would chunk shared code, but letting webpack figure it out leads to more optimal results. Also, using a single output (`lib.min`) as the chunk for the webpack runtime _and_ the `splitChunks` target _and_ a named entry point (old line 130) is no longer allowed by webpack 5. The new configuration is simpler and, IMO, better.